### PR TITLE
Refactoring of neuron classes.

### PR DESCRIPTION
**Pull request by juanchopanza**
Make standard and ezy neurons more uniform. Viewing functions now work for both.

### CODE REVIEW

**Comment by juanchopanza** [comment]
@mgeplf Can you have a look? This is re-factoring. We can leave more fundamental design issues for later.

File: examples/log_neuron.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def point_iter(iterator):
                                        for p in nrn.soma.iter()])
 
     LOG.info('Neuron tree init points, types')
-    for tt in nrn.neurite_trees:
+    for tt in nrn.neurites:
```
Comment (mgeplf) [inline examples/log_neuron.py]: fine w/ a rename, not sure I need to review that

File: neurom/core/neuron.py [hunk 2]
```diff
@@ -156,8 +155,4 @@ def i_neurite(self, iterator_type, mapping=None, tree_filter=None):
             mapping: optional function to apply to the iterator's target.
             tree_filter: optional top level filter on properties of neurite tree objects.
         '''
-        nrt = (self.neurite_trees if tree_filter is None
-               else filter(tree_filter, self.neurite_trees))
-
-        chain_it = chain(*imap(iterator_type, nrt))
-        return chain_it if mapping is None else tr.imap_val(mapping, chain_it)
+        return tr.i_chain(self.neurites, iterator_type, mapping, tree_filter)
```
Comment (mgeplf) [inline neurom/core/neuron.py]: this is just proxy, right?  I would be tempted to have people be explicit, but not a big deal

File: neurom/ezy/neuron.py [hunk 3]
```diff
@@ -109,6 +109,8 @@ class Neuron(object):
     def __init__(self, filename, iterable_type=np.array):
         self._iterable_type = iterable_type
         self._nrn = load_neuron(filename, set_tree_type)
+        self.soma = self._nrn.soma
```
Comment (mgeplf) [inline neurom/ezy/neuron.py]: depending on the type here, I would be concerned that someone might expect to change _nrn, and see show up in soma.  Or is it a container so they point to the same object?

**Comment by mgeplf** [comment]
+1

File: neurom/ezy/neuron.py (hunk 3 above)
Comment (juanchopanza) [inline neurom/ezy/neuron.py]: Same object, and this class owns it.

File: neurom/core/neuron.py (hunk 2 above)
Comment (juanchopanza) [inline neurom/core/neuron.py]: I didn't get that. What do you mean?

Comment (mgeplf) [inline neurom/core/neuron.py]: typing tr.i_chain(self.neurites, ...) vs self.i_neurites_chains(....) is about the same, and is more explicit, imo.  No biggie though

Comment (mgeplf) [inline neurom/core/neuron.py]: OTOH, it's nice to hint at how an interface works, so this makes sense in that context

**Pull request merged**